nucypher/blockchain/eth/token.py

```diff
@@ -275,14 +282,134 @@ def validate_duration(self, raise_on_fail=True) -> Union[bool, Tuple[Tuple[bool,
              .format(minimum=self.economics.minimum_locked_periods, duration=self.duration)),
 
             (self.economics.maximum_locked_periods >= self.duration,
-             'Locktime ({duration}) too long; must be no more than {maximum}'
+             'Stake duration of ({duration}) is too long; must be no more than {maximum} periods.'
              .format(maximum=self.economics.maximum_locked_periods, duration=self.duration)),
         )
 
         if raise_on_fail is True:
             self.__handle_validation_failure(rulebook=rulebook)
         return all(rulebook)
+
+    #
+    # Blockchain
+    #
+
+    @classmethod
+    def __deposit(cls, miner, amount: int, lock_periods: int) -> Tuple[str, str]:
+        """Public facing method for token locking."""
+
+        approve_txhash = miner.token_agent.approve_transfer(amount=amount,
+                                                            target_address=miner.miner_agent.contract_address,
+                                                            sender_address=miner.checksum_public_address)
+
+        deposit_txhash = miner.miner_agent.deposit_tokens(amount=amount,
+                                                          lock_periods=lock_periods,
+                                                          sender_address=miner.checksum_public_address)
+
+        return approve_txhash, deposit_txhash
+
+    def divide(self, target_value: NU, additional_periods: int = None) -> tuple:
+        """
+        Modifies the unlocking schedule and value of already locked tokens.
+
+        This actor requires that is_me is True, and that the expiration datetime is after the existing
+        locking schedule of this miner, or an exception will be raised.
+       """
+
+        # Re-read stakes from blockchain and select stake to divide
+        current_stake = self
+
+        # Ensure selected stake is active
+        if current_stake.is_expired:
+            raise self.StakingError(f'Cannot divide an expired stake')
+
+        if target_value >= current_stake.value:
+            raise self.StakingError(f"Cannot divide stake; Target value ({target_value}) must be less "
+                                    f"than the existing stake value {current_stake.value}.")
+
+        #
+        # Generate SubStakes
+        #
+
+        # Modified Original Stake
+        modified_stake = Stake(miner=self.miner,
+                               index=self.index,
+                               start_period=current_stake.start_period,
+                               end_period=current_stake.end_period,
+                               value=target_value)
+
+        # New Derived Stake
+        end_period = current_stake.end_period + additional_periods
+        new_stake = Stake(miner=self.miner,
+                          start_period=current_stake.start_period,
+                          end_period=end_period,
+                          value=current_stake.value - target_value,
```

Comment:
`current_stake.value - target_value` for old stake, `target_value` for new one

Reply:
:heavy_check_mark: